angel_system/impls/detect_activities/two_stage/two_stage.py

```diff
@@ -23,13 +25,20 @@ def __init__(self, checkpoint: str, num_classes: int):
 
          # FCN: input image (3, 1280, 720) -> Output feature from ResNext (2048)
         self.fcn = SpatialFCNModule('resnext')
-         # Temporal: (Dict) Input feature from ResNext (2048) + Aux_data ->
-         # Output (1, num_classes)
+        # Temporal: (Dict) Input feature from ResNext (2048) + Aux_data ->
+        # Output (1, num_classes)
         self.temporal = RULSTM(num_classes, hidden=128, dropout=0, depth=3)
-
+        check = torch.load(checkpoint)
+        
         self.fcn.eval()
+        
         # Load checkpoint
-        self.temporal.load_state_dict(torch.load(checkpoint))
+        temp_check = dict()
+        for key in check['state_dict'].keys():
+            if 'temporal' in key:
+                new_key = key.split('temporal.')[1]
+                temp_check[new_key] = check['state_dict'][key]
```

Comment:
Are there some additional comments we can add that explain this additional logic?